### PR TITLE
Upgrade (almost) all the deps

### PR DESCRIPTION
**Pull request by halgari**
Upgrades almost all the dependencies in the app that are behind. Does not upgrade a SMAPI dependency (so we stay in sync with SMAPI) or the Bannerlord launcer code (that has a breaking change in a minor release)

### CODE REVIEW

File: Directory.Packages.props [hunk 1]
```diff
@@ -57,21 +57,21 @@
...
     <PackageVersion Include="Microsoft.Extensions.Logging" Version="8.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Logging.Abstractions" Version="8.0.1" />
-    <PackageVersion Include="Microsoft.Extensions.ObjectPool" Version="8.0.4" />
+    <PackageVersion Include="Microsoft.Extensions.ObjectPool" Version="8.0.5" />
     <PackageVersion Include="Microsoft.IO.RecyclableMemoryStream" Version="3.0.0" />
     <PackageVersion Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.4" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.8.0" />
-    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.8.0" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp" Version="4.9.2" />
+    <PackageVersion Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.9.2" />
```
Comment (erri120) [inline Directory.Packages.props:70]: These package versions are tied to the roslyn compiler version

Comment (erri120) [reply Directory.Packages.props:70]: As seen here: https://github.com/Nexus-Mods/NexusMods.App/pull/1064

Comment (halgari) [reply Directory.Packages.props:70]: yep, good point

**Review by erri120 — APPROVED** [review]

**Comment by erri120** [comment]
This is what you usually do with those packages: https://github.com/meziantou/Meziantou.Analyzer/blob/main/Directory.Build.targets

**Pull request merged**